crates/sdk/src/tx.rs

```diff
@@ -2827,6 +2828,63 @@ pub async fn build_ibc_transfer(
         masp_fee_data
     };
 
+    if let Some(TxTransparentTarget {
+        target,
+        token,
+        amount,
+    }) = &args.frontend_sus_fee
+    {
+        match (&source, &args.ibc_shielding_data) {
+            (&MASP, None) => {
+                // Validate the amount given
+                let validated_amount = validate_amount(
+                    context,
+                    amount.to_owned(),
+                    token,
+                    args.tx.force,
+                )
+                .await?;
+                masp_transfer_data.sources.push((
+                    args.source.clone(),
+                    args.token.clone(),
+                    validated_amount,
+                ));
+                masp_transfer_data.targets.push((
+                    TransferTarget::Address(target.to_owned()),
+                    token.to_owned(),
+                    validated_amount,
+                ));
+
+                transfer = transfer
+                    .transfer(
+                        source.to_owned(),
+                        target.to_owned(),
+                        token.to_owned(),
+                        validated_amount,
+                    )
+                    .ok_or(Error::Other(
+                        "Combined transfer overflows".to_string(),
```

Comment:
Will this catch the case that both the sus fee + raw transfer amount together exceed source's balance?